bin/problem.py

```diff
@@ -916,38 +918,104 @@ def validate_data(problem, mode: validate.Mode, constraints: dict | bool | None
                 log(f"Not running answer_validators for{msg} problems.")
             return True
 
+        action = (
+            "Invalidation"
+            if mode == validate.Mode.INVALID
+            else (
+                f"Collecting {mode} constraints" if constraints else f"{mode} validation"
+            ).capitalize()
+        )
+
+        testcases = problem.testcases(mode=mode)
+        return problem._validate_data(mode, constraints, action, testcases)
+
+    def validate_invalid_extra_data(p) -> bool:
+        base_path = p.tmpdir / "invalid_data"
+        # find at least one (valid?) testcase to modify
+        test_paths = list(glob(p.path, "data/sample/**/*.in"))
+        test_paths += list(glob(p.path, "data/secret/**/*.in"))
+        test_paths = [path for path in test_paths if path.with_suffix(".ans").exists()]
+        test_paths.sort()
+        test_path = test_paths[0] if test_paths else None
+        if test_path:
+            test_case = test_path.relative_to(p.path / "data").with_suffix("")
+            log(f"Generating invalid testcases based on: {test_case}")
+        debug(f"writing generated invalid testcases to: {base_path}")
```

Comment:
This debug log is visible, even without `-v`, I guess that's intended ('cause the log is "debug" and not "verbose")?